src/server/ua_subscription_alarms_conditions.c

```diff
@@ -2612,6 +2661,144 @@ UA_StatusCode UA_Server_deleteCondition(UA_Server *server, const UA_NodeId condi
     return UA_Server_deleteNode(server, condition, true);
 }
 
+static UA_StatusCode getLowLimit(UA_Server *server, UA_NodeId conditionId, UA_Double *lowLimit) {
+    UA_Variant *value = UA_Variant_new();
+    UA_StatusCode retval = UA_Server_readObjectProperty(server, conditionId,
+                                UA_QUALIFIEDNAME(0, CONDITION_FIELD_LOWLIMIT),
+                                value);
+    *lowLimit = *(UA_Double*) value->data;
+    UA_Variant_delete(value);
+    return retval;
+}
+
+static UA_StatusCode getLowLowLimit(UA_Server *server, UA_NodeId conditionId, UA_Double *lowLowLimit) {
+    UA_Variant *value = UA_Variant_new();
```

Comment:
can't this be done on the stack. the life-time of the variant is well-defined and short. no need to use the heap?

so also the other cases